presto-spi/src/main/java/com/facebook/presto/spi/statistics/ColumnStatistics.java

```diff
@@ -114,6 +114,13 @@ public static Builder builder()
         return new Builder();
     }
 
+    /**
+     * If one of the estimates below is unspecified, the default "unknown" estimate value
+     * (represented by floating point NaN value) may cause the resulting symbol statistics
+     * to be "unknown" as well.
+     * Make sure to specify all the estimates below to allow cost-based query optimization.
```

Comment:
This comment implies that cost base optimization can only happen when all estimates are provided. If that's true, why do we allow setting them individually? What happens if, say, `dataSize` is missing, but the others are present?

Reply:
@electrum If I understand correctly, in case `dataSize` is missing - the cost won't be estimated, since the `averageRowSize` below will be NaN:
https://github.com/prestodb/presto/blob/f112441aa3f4e6c485f5333bd4f38901c0f96433/presto-main/src/main/java/com/facebook/presto/cost/TableScanStatsRule.java#L79

Is it possible to verify that the user sets all the estimates, or none of them at `Builder.build()`?
https://github.com/prestodb/presto/blob/b6addd420c5bec97e675eb5f4c238301cf80c083/presto-spi/src/main/java/com/facebook/presto/spi/statistics/ColumnStatistics.java#L154-L157

Reply:
If we don't have dataSize it will be unknown in the symbol stats estimate, but we use a default value for cost estimations. For fixed-width types (e.g. int, boolean) we use the sizes of the data type assuming no compression and for non-fixed-width types (e.g. varchar, varbinary) we use a default value of 50 bytes.

Reply:
(So basically the code comment itself incorrect, but your impression of the consequences may be)

Reply:
@rschlussel Thanks for the clarification, I have remove the last line from the comment, and pushed 1627b3d7389a77c0dceb664f29306731a6b09e10.